src/fastfeedparser/main.py

```diff
@@ -44,640 +47,573 @@ def parse(source):
-                "title",
-                "{http://www.w3.org/2005/Atom}title",
-                "{http://purl.org/rss/1.0/}channel/{http://purl.org/rss/1.0/}title",
-            ),
-            (
-                "link",
-                "link",
-                "{http://www.w3.org/2005/Atom}link",
-                "{http://purl.org/rss/1.0/}channel/{http://purl.org/rss/1.0/}link",
-                True,
-            ),
-            (
-                "subtitle",
-                "description",
-                "{http://www.w3.org/2005/Atom}subtitle",
-                "{http://purl.org/rss/1.0/}channel/{http://purl.org/rss/1.0/}description",
-            ),
-            (
-                "generator",
-                "generator",
-                "{http://www.w3.org/2005/Atom}generator",
-                "{http://purl.org/rss/1.0/}channel/{http://webns.net/mvcb/}generatorAgent",
-            ),
-            (
-                "publisher",
-                "publisher",
-                "{http://www.w3.org/2005/Atom}publisher",
-                "{http://purl.org/rss/1.0/}channel/{http://purl.org/dc/elements/1.1/}publisher",
-            ),
-            (
-                "author",
-                "author",
-                "{http://www.w3.org/2005/Atom}author/{http://www.w3.org/2005/Atom}name",
-                "{http://purl.org/rss/1.0/}channel/{http://purl.org/dc/elements/1.1/}creator",
-            ),
-            (
-                "updated",
-                "lastBuildDate",
-                "{http://www.w3.org/2005/Atom}updated",
-                "{http://purl.org/rss/1.0/}channel/{http://purl.org/dc/elements/1.1/}date",
-            ),
-        ]
+    feed = _parse_feed_info(channel, feed_type)
 
-        for field in fields:
-            value = get_feed_value(*field[1:])
-            if value:
-                feed[field[0]] = value
-
-        # Add title_detail and subtitle_detail
-        if "title" in feed:
-            feed["title_detail"] = {
-                "type": "text/plain",
-                "language": channel.get("{http://www.w3.org/XML/1998/namespace}lang"),
-                "base": channel.get("{http://www.w3.org/XML/1998/namespace}base"),
-                "value": feed["title"],
-            }
-        if "subtitle" in feed:
-            feed["subtitle_detail"] = {
-                "type": "text/plain",
-                "language": channel.get("{http://www.w3.org/XML/1998/namespace}lang"),
-                "base": channel.get("{http://www.w3.org/XML/1998/namespace}base"),
-                "value": feed["subtitle"],
-            }
+    # Parse entries
+    entries: list[FastFeedParserDict] = []
+    feed["entries"] = entries
+    for item in items:
+        entry = _parse_feed_entry(item, feed_type)
+        # Ensure that titles and descriptions are always present
+        entry["title"] = entry.get("title", "").strip()
+        entry["description"] = entry.get("description", "").strip()
+        entries.append(entry)
```

Comment:
BREAKING: summary was removed because it's always empty. Better not to return weird data.